### ISSUE TITLE
[RyuJIT/ARM32] Assertion failed 'this->OperIsBinary()'

### ISSUE THREAD

**Issue by sjsinju** [dotnet/coreclr#13049]
in Pr0 test
```
Assertion failed 'this->OperIsBinary()'

    File: /opt/code/github/hqueue/coreclr/src/jit/gentree.cpp Line: 5595
```

Failed tests
```
GC/Features/HeapExpansion/bestfit-finalize/bestfit-finalize.sh
JIT/jit64/hfa/main/testA/hfa_nf0A_r/hfa_nf0A_r.sh
JIT/jit64/hfa/main/testA/hfa_nf1A_d/hfa_nf1A_d.sh
JIT/jit64/hfa/main/testA/hfa_nf1A_r/hfa_nf1A_r.sh
JIT/jit64/hfa/main/testB/hfa_nf0B_d/hfa_nf0B_d.sh
JIT/jit64/hfa/main/testB/hfa_nf0B_r/hfa_nf0B_r.sh
```

**Comment by sjsinju** [comment]
@BruceForstall  Please assign this to me.

**Comment by sjsinju** [comment]
Fixed with #13050
